### PR TITLE
Add list functionality to rich text editor

### PR DESCRIPTION
**Pull request by alunturner**
This PR has design approval. The new code:

- adds buttons to toggle bulleted and numbered lists on and off
- adds icons for those buttons
- adds some css to make the composer input look like the timeline display
- adds some css to hide the linebreak that is always present at the end of the rust model output - if we don't hide this linebreak then it is apparent when we have a list element in the composer, as we see an empty line after the list items and this looks bad
- adds tests for the new buttons, refactor test to make them easier to expand for new buttons when they are added

Images, showing composer input and timeline input above:
List item only:
Element___Alun_Turner

Text then list item:
Element___Alun_Turner

Text then list item then text:
Element___Alun_Turner

Example of ordered list
Element___Alun_Turner

Example of adjacent list display:
Element___Alun_Turner

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Add list functionality to rich text editor ([\#9871](https://github.com/matrix-org/matrix-react-sdk/pull/9871)). Contributed by @alunturner.

### CODE REVIEW

**Review by andybalaam — APPROVED** [review]
Woohoo!

**Comment by alunturner** [comment]
A few changes post approval due to design feedback:

- changes the height at which border radius automatically changes on the composer to account for the new line height of the composer lines
- changes bullet points to be discs, not circles
- fixes regression that caused shift + enter to stop inserting a newline and send the message instead

**Comment by andybalaam** [comment]
Still approved, but you may need to update to latest and check carefully because a big typescript strictness change just came through :-)

File: res/css/views/rooms/wysiwyg_composer/components/_Editor.pcss [hunk 1]
```diff
@@ -25,6 +25,7 @@ limitations under the License.
     }
 
     .mx_WysiwygComposer_Editor_content {
+        line-height: $font-22px;
```
[the remark this answers is not in the document]
Comment (alunturner) [reply res/css/views/rooms/wysiwyg_composer/components/_Editor.pcss:28]: Have checked - still remains at 40px.

**Pull request merged**